### ISSUE TITLE
Performance Monitoring ASP.NET Core Auto instrumentation

### ISSUE THREAD

**Issue by bruno-garcia** [getsentry/sentry-dotnet#716]
This is a follow up of #715.

The goal is to ship GA of ASP.NET Core integration for performance monitoring.

The scope is requests into the process automatically capture a transaction to Sentry with timing information for the overall request. Additional spans will be added in follow up issues.

**Comment by bruno-garcia** [comment]
@kanadaj raised some points about the current instrumentation on ASP.NET Core. We could add a second middleware just for performance monitoring btw, and inject it at the right place (after UseRouting or UseMVC are called). [Seems Andrew Lock blogged about this](https://andrewlock.net/inserting-middleware-between-userouting-and-useendpoints-as-a-library-author-part-1/).

We need to be able to add our middleware to start the transaction as early as possible but only once the route template exists. Addiing the transaction before that would require a mutable transaction and the sampling decision callback which runs when the transaction starts wouldn't have that information, which is the most important (I want to sample out `/health-check` for example).

**Comment by kanadaj** [comment]
Small correction, should be either after `UseRouting` or `UseEndpoints`.

Additionally, legacy health checks don't need routing, so they may be inserted before `UseRouting`.

On the topic of inserting a middleware with `IStartupFilter`, I propose to either add a string option for the full name of the middleware to insert the middleware after, a bool predicate for the same purpose, or a simple bool to decide whether the middleware is inserted after `UseRouting`/`UseEndpoints` or at the very start of the pipeline in case of a custom middleware pipeline.

**Comment by Tyrrrz** [comment]
@bruno-garcia 

>The scope is requests into the process automatically capture a transaction to Sentry with timing information for the overall request. Additional spans will be added in follow up issues.

This seems to be already done, except for additional spans. How should we proceed?

**Comment by bruno-garcia** [comment]
It's out on 3.0.0 in preview, will GA soon y'all 🥳 

Thanks @Tyrrrz for all the work!
